draft-irtf-cfrg-aegis-aead.md

```diff
@@ -1580,26 +1572,44 @@ With AEGIS-128L and AEGIS-128X, random nonces can safely encrypt up to 2<sup>48<
 
-Each variant can be used as a MAC by calling the `Encrypt()` function with the message as the `ad` and leaving `msg` empty, resulting in just a tag. However, they MUST NOT be used as a hash function; if the key is known, inputs generating state collisions can easily be crafted. Similarly, as opposed to hash-based MACs, tags MUST NOT be used for key derivation as there is no proof they are uniformly random.
+## Security Guarantees
 
-As shown in {{D23}}, AEGIS-128X and AEGIS-256X share the same security properties and requirements as AEGIS-128L and AEGIS-256 respectively. In particular, the security level and usage limits remain the same.
+AEGIS-256 offers 256-bit message security against plaintext and state recovery, whereas AEGIS-128L offers 128-bit security.
+
+Under the assumption that the secret key is unknown to the attacker, all AEGIS variants target 128-bit security against forgery attacks regardless of the tag size.
 
 Without the ability to set the associated data, a successful forgery doesn't increase the probability of subsequent forgeries. Reforgeability reilience of AEGIS is analyzed in {{FLLW17}}.
 
-The security of AEGIS against timing and physical attacks is limited by the implementation of the underlying `AESRound()` function. Failure to implement `AESRound()` in a fashion safe against timing and physical attacks, such as differential power analysis, timing analysis or fault injection attacks, may lead to leakage of secret key material or state information. The exact mitigations required for timing and physical attacks also depend on the threat model in question.
+AEGIS-128X and AEGIS-256X share the same security properties and requirements as AEGIS-128L and AEGIS-256 respectively. In particular, the security level and usage limits remain the same. {{D23}}
 
 AEGIS is considered secure against guess-and-determine attacks aimed at recovering the state from observed ciphertexts. This resilience extends to quantum adversaries in the Q1 model, wherein quantum attacks do not confer any practical advantage for decrypting previously recorded ciphertexts or achieving key recovery.
 
 Security analyses of AEGIS can be found in {{AEGIS}}, {{M14}}, {{ENP19}}, {{LIMS21}}, {{JLD21}}, {{STSI23}}, {{IR23}}, {{BS23}} and {{FLLW17}}.
 
+## Implementation Security
+
+The security of AEGIS against timing and physical attacks is limited by the implementation of the underlying `AESRound()` function. Failure to implement `AESRound()` in a fashion safe against timing and physical attacks, such as differential power analysis, timing analysis or fault injection attacks, may lead to leakage of secret key material or state information. The exact mitigations required for timing and physical attacks also depend on the threat model in question.
+
+## Key Commitment
```

Comment:
I moved these under `Usage Guidelines` but then we need to organize and find subtitles for the rest of the content in that section.

And I'm a little bit stuck here. What would you suggest?

Reply:
My original thinking was not to have subtitles under `Usage Guidelines` so you just have 3 subsections for `Security Considerations` - usage, implementation, security. However, there's then a bit more text in one subsection.

I'm also not sure what other subtitles could be used and not having subtitles for that first part but keeping `Key Commitment`/`Multi-User Security` there might look a bit odd.

Reply:
Here's an attempt.

"Additional Considerations" is very generic, but I couldn't find anything better.

Reply:
The paragraph on the parallelism degree affecting the output was removed. It feels at odd with the rest of that section, and even with the entire "Security Considerations" section.

Reply:
Only alternative I can think of is `Parameter Selection`, but that goes into associated data too and overlaps with key commitment. I think those headings are ok unless you prefer that, although should we lowercase the 'and' - `Key and Nonce Selection`?

Reply:
Alright, let's keep the current headings, then.

The paragraph may fit into the `Parallel Modes` intro. But unless you feel like it's important to keep it, I'd remove it entirely.

Reply:
I do think the `And` should be `and` based on [RFC 7322](https://datatracker.ietf.org/doc/html/rfc7322#section-3.4):

> Typically, all words in a title will be capitalized, except for internal articles, prepositions, and conjunctions.

In response to this...

> The paragraph may fit into the Parallel Modes intro. But unless you feel like it's important to keep it, I'd remove it entirely.

I would probably mention it in the `Parallel Modes` intro just for clarity, although it's identifiable if you implement them or look at the different test vectors and notice the same inputs.

The associated data affecting the ciphertext seems like a fun fact to keep in because that's not what traditionally happens, but I don't know where that would go unless you put it in the `Introduction` like the discussion of the key erasure.

Reply:
> I would probably mention it in the Parallel Modes intro just for clarity, although it's identifiable if you implement them or look at the different test vectors and notice the same inputs.

How about:

For a given `key` and `nonce` pair, varying the parallelism degree results in distincts key streams > and authentication tags. Thus, reusing a key across different variants does not pose a security concern.

?

Reply:
> The associated data affecting the ciphertext seems like a fun fact to keep in because that's not what traditionally happens, but I don't know where that would go unless you put it in the Introduction like the discussion of the key erasure.

This is the case with all sponge-based constructions :)

That fun fact might be understood as "it's okay to reuse key/nonce pairs if ADs are different", which is not universally true in AEGIS. So maybe we shouldn't reintroduce this.

Reply:
or... how about this, in the "Security Guarantees" section:

> Encrypting a message with the same key and nonce, but with different additional data, generates distinct ciphertexts that do not reveal any additional information about the message.

?

Reply:
> For a given key and nonce pair, varying the parallelism degree results in distincts key streams and authentication tags. Reusing a key across different variants does not pose a security concern.

That's good, although typo with `distincts key streams`. Should be `distinct keystreams`. Just in case you copy and paste.

Reply:
> This is the case with all sponge-based constructions :)
> 
> That fun fact might be understood as "it's okay to reuse key/nonce pairs if ADs are different", which is not universally true in AEGIS. So maybe we shouldn't reintroduce this.

Yeah, but basically no sponge-based AEAD constructions are currently used in practice aside from Ascon. I agree we do need to be careful.

> Encrypting a message with the same key and nonce, but with different additional data, generates distinct ciphertexts that do not reveal any additional information about the message.

I quite like that. I'd slightly tweak it to:

> Encrypting a message with the same key and nonce but different additional data generates distinct ciphertexts that do not reveal any additional information about the message.

Reply:
Great! Let's do that :)